### DISCUSSION TITLE
Generic OAuth2认证方式登录

### DISCUSSION THREAD

**Discussion by TerryWang6** [requarks/wiki#5693] [Help / Questions]
各位大佬下午好，这是我在程序中的配置
![image](https://user-images.githubusercontent.com/92418312/192982296-5ac903be-5ebb-45c9-85a6-e88b6ed7d961.png)
但是程序报错，初步排查是因为点击Generic OAuth2方式登录的时候，跳转URL的字段不正确，我想知道这个字段怎么修改一下名称或者怎么新增字段。
![image](https://user-images.githubusercontent.com/92418312/192983209-fe6dbb27-0032-4797-bf7c-45eb29f0a573.png)
当前的URL：https://open.weixin.qq.com/connect/oauth2/authorize?response_type=code&redirect_uri=http%3A%2F%2F172.17.233.72%3A3035%2Flogin%2Fc1f97b18-c299-4549-a44a-aeac9c78cdf6%2Fcallback&client_id=ww077b3a12b6d224**
期望的URL：https://open.weixin.qq.com/connect/oauth2/authorize?response_type=code&redirect_uri=http%3A%2F%2F172.17.233.72%3A3035%2Flogin%2Fc1f97b18-c299-4549-a44a-aeac9c78cdf6%2Fcallback&appid=ww077b3a12b6d224**
就是最后一个字段名称怎么更换。有懂得的大佬帮忙看一下，谢谢！

**Comment by cdn0x12** [comment]
Looks like you are integrating WeCom SSO to your wiki instance. As you mentioned WeCom uses `appid` as its `client_id` and requires additional params that don't comply with the Generic OAuth2 standard, so it might be necessary to implement another module specifically for the WeCom login workflow.
